d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -547,6 +568,8 @@ static ResourceType fromTypeUrl(String typeUrl)
     }
   }
 
+  public abstract void start();
```

Comment:
Having both `start` and `startRpcStream` feels dangerous. What happens if you call one but not the other? Are you supposed to call both? I know that `startRpcStream` is a bad name, and it should likely have been just `start` from the beginning, but I think it is too late to change this now.

Reply:
since "_xdsClientJmx.setXdsClient(this);" has to be done in XdsClientImpl. If we put it into another public method, and if there are more things to be done at startup for xdsClientimpl we put them into different public methods, then external class like XdsToD2PropertiesAdaptor need to call them all. It feels dirty and odd.

Currently only XdsToD2PropertiesAdaptor is calling startRpcStream, I wonder if we could start using start() from now on, and just avoid calling startRpcStream from external class. I can put a comment on startRpcStream to note that.

Or do you have a better way to solve this?